Drivers/Src/Display.c

```diff
@@ -1,325 +1,280 @@
- */
-static void sendStartOfAssignment(enum CommandString_t obj_index, enum CommandString_t attr_index)
-{
-    char *obj = CommandStrings[obj_index];
-    char *attr = CommandStrings[attr_index];
-    int len = strlen(obj);
-
-    if (len != 0)
-    { // If not global
-        BSP_UART_Write(DISP_OUT, obj, len);
-        BSP_UART_Write(DISP_OUT, (char *)DELIMITER, strlen(DELIMITER));
-    }
-
-    BSP_UART_Write(DISP_OUT, attr, strlen(attr)); // Send the attribute
-    BSP_UART_Write(DISP_OUT, (char *)ASSIGNMENT, 1);
-}
+ErrorStatus Display_PutNext(Display_Cmd_t cmd){
+	CPU_TS ticks;
+	OS_ERR err;
 
-/**
- * Sends out a string of the form "obj.attr=msg" or "attr=msg"
- * Use to update string fields of display objects
- */
-static ErrorStatus updateStringValue(enum CommandString_t obj_index, enum CommandString_t attr_index, char *msg)
-{
-    sendStartOfAssignment(obj_index, attr_index);
-
-    BSP_UART_Write(DISP_OUT, msg, strlen(msg));
-    BSP_UART_Write(DISP_OUT, (char *)TERMINATOR, strlen(TERMINATOR));
-
-    // Get a response from the display
-    char buf[8];
-    BSP_UART_Read(DISP_OUT, buf);
-    return (IsNextionFailure(buf[0])) ? ERROR : SUCCESS;
-}
+	OSMutexPend(&DisplayQ_Mutex, 0, OS_OPT_POST_NONE, &ticks, &err);
+  assertOSError(OS_DISPLAY_LOC, err);  
+	
+	bool success = disp_fifo_put(&msg_queue, cmd);
 
-/**
- * Sends out a string of the form "obj.attr=val" or "attr=val"
- * Use to update integer fields of display objects
- */
-static ErrorStatus updateIntValue(enum CommandString_t obj_index, enum CommandString_t attr_index, int32_t val)
-{
-    sendStartOfAssignment(obj_index, attr_index);
+	OSMutexPost(&DisplayQ_Mutex, OS_OPT_POST_NONE, &err);
+	assertOSError(OS_DISPLAY_LOC, err);
 
-    char number[12]; // To store converted int
-    sprintf(number, "%ld", val);
+	if(success){
+		OSSemPost(&DisplayQ_Sem4, OS_OPT_POST_NONE, &err);
+		assertOSError(OS_DISPLAY_LOC, err);
+	}
 
-    BSP_UART_Write(DISP_OUT, number, strlen(number));
-    BSP_UART_Write(DISP_OUT, (char *)TERMINATOR, strlen(TERMINATOR));
+	return success ? SUCCESS : ERROR;
+}
 
-    // Get a response from the display
-    char buf[8];
-    BSP_UART_Read(DISP_OUT, buf);
-    return (IsNextionFailure(buf[0])) ? ERROR : SUCCESS;
+ErrorStatus Display_SendNext(){
+	Display_Cmd_t cmd;
+
+	OS_ERR err;
+	CPU_TS ticks;
+    
+	OSSemPend(&DisplayQ_Sem4, 0, OS_OPT_PEND_BLOCKING, &ticks, &err);
+	assertOSError(OS_DISPLAY_LOC, err);
+	
+	OSMutexPend(&DisplayQ_Mutex, 0, OS_OPT_POST_NONE, &ticks, &err);
+	assertOSError(OS_DISPLAY_LOC, err);
+	
+	bool result = disp_fifo_get(&msg_queue, &cmd);
+	OSMutexPost(&DisplayQ_Mutex, OS_OPT_POST_NONE, &err);
+	assertOSError(OS_SEND_CAN_LOC, err);
+	
+	if(result == ERROR)
+		return ERROR;
+	
+	char msg[32];
+	if((cmd.numArgs == 1 && cmd.args != NULL && cmd.argTypes != NULL) && cmd.op != NULL && cmd.attr != NULL){	// Assignment commands have only 1 arg, an operator, and an attribute
```

Comment:
The extra set of parentheses isn't technically necessary.

Reply:
Right, it was more for readability (to group the argument stuff together)